docs/api/io/BufferedSerial.md

```diff
@@ -0,0 +1,71 @@
+
+To transmit multiple bytes, the class uses an intermediary buffer to store the bytes to send and monitors the serial interface to transfer them to the hardware buffer as soon as it is available. However, all bytes are written unbuffered if in a critical section.
+Using intermediary buffers allows it to be used reliably for input from non-interrupt context whilst avoiding excess spinning waiting for transmission buffer space.
+
+The RX and TX buffers are circular buffers with pre-allocated sizes configurable using [`drivers/mbed_lib.json`](https://github.com/ARMmbed/mbed-os/blob/master/drivers/mbed_lib.json).
+
+## Configuration
+
+The following parameters can be configured at object instantiation:
+
+  - _TX_
+  - _RX_
+  - _Baud rate_ - defaults to the value configured in `mbed-os/platform/mbed_lib.json`.
+
+The following parameters can be configured after an `BufferedSerial` object instantiation.
+
+  - _Baud rate_
+  - _Data frame length_
+  - _Parity bit_
+  - _Stop bits_
+
+The default settings for a microcontroller are described as _9600-8-N-1_, a common notation for serial port settings.
+
+Additionally, hardware flow control can also be configured if necessary.
+
+You can view more information about the configurable settings and functions in the class reference.
+
+## Class reference
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/mbed-os/development/mbed-os-api-doxy/classmbed_1_1_buffered_serial.html)
+
+<span class="notes">**Note**: On a Windows machine, you need to install a USB serial driver. See [Windows serial configuration](../tutorials/serial-communication.html#windows-serial-driver).</span>
+
+## Example
+
+[![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed_example/code/BufferedSerial/)](https://os.mbed.com/teams/mbed_example/code/BufferedSerial/file/112a40a5991a/main.cpp)
+
+
+## Printing to the console
+
+Mbed OS redefines target-dependent I/O functions in the C library to allow the C standard I/O library functions (`s\v\f\n\printf`, `scanf`, etc) to be used in user application for printing to the console.
+
+The [system I/O retarget code](https://github.com/ARMmbed/mbed-os/blob/master/platform/source/mbed_retarget.cpp) can be configured to be buffered or unbuffered depending on the configuration in [`platform/mbed_lib.json`](https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_lib.json). When it is buffered, [an instance of a `BufferedSerial` class](https://github.com/ARMmbed/mbed-os/blob/master/platform/source/mbed_retarget.cpp#L340) is used by the retarget code to perform the actual printing. This is where `BufferedSerial`'s `Filehandle` inheritance is used.
```

Comment:
I am not keen on having so many links. It might become a maintenance nightmare. 
Also, I would list the configuration parameters from `platform/mbed_lib.json` that are applicable to thsi section.